### PR TITLE
Update cpp.md

### PR DESCRIPTION
**Pull request by elee0625**
### What does this PR do?
C++ Public Docs Modification

### Motivation
Onboarding Initiaitve

### Additional Notes
Content switches were pre-approved by D.Goffredo in an offline sync.
---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by cgilmour — COMMENTED** [review]
This looks reasonable to me and doesn't affect technical aspects of C++ tracing, just makes overall descriptions of tracing setup more consistent.

Someone else would be better to review specific wording warnings and errors - it's not something I'm the appropriate person to judge.

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 1]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
+{{% tab "Containers" %}}
+
+1. Set `apm_non_local_traffic: true` in the `apm_config` section of your main [`datadog.yaml` configuration file][1].
+
+2. See the specific setup instructions to ensure that the Agent is configured to receive traces in a containerized environment:
+
+{{< partial name="apm/apm-containers.html" >}}
+</br>
+
+3. After having instrumented your application, the tracing client sends traces to `localhost:8126` by default.  If this is not the correct host and port change it by setting the below env variables:
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:51]:
```suggestion
3. The tracing client sends traces to `localhost:8126` by default. If this is not the correct host and port for your Agent, set the `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT` environment variables.
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 2]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
 
-Compile against [OpenTracing-cpp][6].
+- Step-by-step instructions scoped to your deployment configuration (hosts, Docker, Kubernetes, or Amazon ECS).
+- Dynamically set `service`, `env`, and `version` tags.
+- Enable features such as ingesting 100% of traces, and Trace ID injection into logs during setup.
 
-## Installation
+## Configure the Datadog Agent for APM
+
+Install and configure the Datadog Agent to receive traces from your now instrumented application. By default the Datadog Agent is enabled in your `datadog.yaml` file under `apm_config` with `enabled: true` and listens for trace traffic at `localhost:8126`. For containerized environments, follow the links below to enable trace collection within the Datadog Agent.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:39]:
```suggestion
Install and configure the Datadog Agent to receive traces from your instrumented application. By default the Datadog Agent is enabled in your `datadog.yaml` file under `apm_config` with `enabled: true` and listens for trace traffic at `localhost:8126`. For containerized environments, follow the links below to enable trace collection within the Datadog Agent.
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 3]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
+1. Set `apm_non_local_traffic: true` in the `apm_config` section of your main [`datadog.yaml` configuration file][1].
+
+2. See the specific setup instructions to ensure that the Agent is configured to receive traces in a containerized environment:
+
+{{< partial name="apm/apm-containers.html" >}}
+</br>
+
+3. After having instrumented your application, the tracing client sends traces to `localhost:8126` by default.  If this is not the correct host and port change it by setting the below env variables:
+
+`DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT`.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:53]:
```suggestion
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 4]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
+
+To connect to the agent using Unix Domain Sockets, `DD_TRACE_AGENT_URL` can be used instead. The value should match the Agent's value for `DD_APM_RECEIVER_SOCKET`.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:55]:
```suggestion
To connect to the Agent using Unix Domain Sockets, use `DD_TRACE_AGENT_URL` instead. Set it to the same value as the Agent's value for `DD_APM_RECEIVER_SOCKET`.
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 5]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
+{{% tab "AWS Lambda" %}}
+
+To set up Datadog APM in AWS Lambda, see the [Tracing Serverless Functions][1] documentation.
+
+
+[1]: /tracing/serverless_functions/
+{{% /tab %}}
+{{% tab "Other Environments" %}}
+
+Tracing is available for a number of other environments, such as  [Heroku][1], [Cloud Foundry][2], [AWS Elastic Beanstalk][3], and [Azure App Service][4].
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:69]:
```suggestion
Tracing is available for other environments, including [Heroku][1], [Cloud Foundry][2], [AWS Elastic Beanstalk][3], and [Azure App Service][4].
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 6]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
+
+For other environments, please refer to the [Integrations][5] documentation for that environment and [contact support][6] if you are encountering any setup issues.
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:71]:
```suggestion
For other environments, see the [Integrations][5] documentation for that environment and [contact support][6] if you encounter setup issues.
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 7]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
+[1]: /agent/basic_agent_usage/heroku/#installation
+[2]: /integrations/cloud_foundry/#trace-collection
+[3]: /integrations/amazon_elasticbeanstalk/
+[4]: /infrastructure/serverless/azure_app_services/#overview
+[5]: /integrations/
+[6]: /help/
+{{% /tab %}}
+{{< /tabs >}}
+
+## Instrument your Application
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:82]:
```suggestion
## Instrument your application
```

File: content/en/tracing/trace_collection/dd_libraries/cpp.md [hunk 8]
```diff
@@ -24,17 +24,64 @@ further_reading:
...
 
-Datadog tracing can be enabled in one of two ways:
+After the agent is installed, Datadog tracing can be enabled in one of two ways:
```
Comment (kayayarai) [inline content/en/tracing/trace_collection/dd_libraries/cpp.md:84]:
```suggestion
After the Agent is installed, follow these steps to add the Datadog tracing library to your C++ applications in one of two ways:
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
edit pass

**Review by kayayarai — APPROVED** [review]

**Pull request merged**